safe/report/map_report.py

```diff
@@ -0,0 +1,434 @@
+import logging
+
+# noinspection PyUnresolvedReferences
+from qgis.core import (
+    QgsComposition,
+    QgsRectangle,
+    QgsMapLayer,
+    QgsLayerTreeGroup)
+
+from safe.defaults import disclaimer
+from safe.common.utilities import temp_dir, unique_filename
+from safe.common.version import get_version
+from safe.common.exceptions import (
+    KeywordNotFoundError, LoadingTemplateError)
+from safe.utilities.keyword_io import KeywordIO
+from safe.utilities.resources import resources_path
+from safe.utilities.gis import qgis_version
+from safe.defaults import (
+    default_organisation_logo_path,
+    default_north_arrow_path)
+from safe.report.template_composition import TemplateComposition
+
+LOGGER = logging.getLogger('InaSAFE')
+
+
+class MapReport(object):
+    """A class for creating report using QgsComposition."""
+    def __init__(self, iface, template, layer):
+        """Constructor for the Composition Report class.
+
+        :param iface: Reference to the QGIS iface object.
+        :type iface: QgsAppInterface
+
+        :param template: The QGIS template path.
+        :type template: str
+        """
+        LOGGER.debug('InaSAFE Map class initialised')
+
+        self.iface = iface
+        self.keyword_io = KeywordIO()
+        self.page_dpi = 300.0
+
+        self._template = template
+        self._layer = layer
+        self._template_composition = None
+        self._extent = self.iface.mapCanvas().extent()
+        self._safe_logo = resources_path(
+            'img', 'logos', 'inasafe-logo-url.svg')
+        self._org_logo = default_organisation_logo_path()
+        self._north_arrow = default_north_arrow_path()
+        self._disclaimer = disclaimer()
+        self._template_composition = TemplateComposition(
+            template_path=self.template,
+            map_settings=self.iface.mapCanvas().mapSettings())
+
+    @property
+    def template(self):
+        """Getter to the template"""
+        return self._template
+
+    @template.setter
+    def template(self, template):
+        """Set the template.
+
+        :param template: The template.
+        :type template: str
+        """
+        self._template = template
+
+    @property
+    def layer(self):
+        """Getter to layer that will be used for stats, legend, reporting."""
+        return self._layer
+
+    @layer.setter
+    def layer(self, layer):
+        """Set the layer that will be used for stats, legend and reporting.
+
+        :param layer: Layer that will be used for stats, legend and reporting.
+        :type layer: QgsMapLayer, QgsRasterLayer, QgsVectorLayer
+        """
+        self._layer = layer
+
+    @property
+    def composition(self):
+        """Getter to QgsComposition instance."""
+        return self._template_composition.composition
+
+    @property
+    def extent(self):
+        """Getter to extent for map component in composition."""
+        return self._extent
+
+    @extent.setter
+    def extent(self, extent):
```

Comment:
What about some sanity checking like you have done for organisation_logo. We can check that the extent is valid, not None etc